src/assets/wise5/components/draw/draw-student/draw-student.component.ts

```diff
@@ -0,0 +1,437 @@
+  height: number = 600;
+  latestConnectedComponentState: any;
+  parentStudentWorkIds: number[] = null;
+  isResetButtonVisible: boolean = true;
+  showCopyPublicNotebookItemButton: boolean = false;
+  width: number = 800;
+
+  constructor(
+    protected AnnotationService: AnnotationService,
+    protected ComponentService: ComponentService,
+    protected ConfigService: ConfigService,
+    private DrawService: DrawService,
+    protected NodeService: NodeService,
+    protected NotebookService: NotebookService,
+    private ProjectService: ProjectService,
+    protected StudentAssetService: StudentAssetService,
+    protected StudentDataService: StudentDataService,
+    protected upgrade: UpgradeModule,
+    protected UtilService: UtilService
+  ) {
+    super(
+      AnnotationService,
+      ComponentService,
+      ConfigService,
+      NodeService,
+      NotebookService,
+      StudentAssetService,
+      StudentDataService,
+      upgrade,
+      UtilService
+    );
+  }
+
+  ngOnInit(): void {
+    super.ngOnInit();
+    this.drawingToolId = this.DrawService.getDrawingToolId(this.nodeId, this.componentId);
+    this.showCopyPublicNotebookItemButton = this.ProjectService.isSpaceExists('public');
+    if (this.isDisabled) {
+      this.isResetButtonVisible = false;
+    }
+    this.subscribeToNotebookItemChosen();
+  }
+
+  ngAfterViewInit(): void {
+    this.initializeDrawingTool();
+  }
+
+  handleNodeSubmit(): void {
+    this.submit('nodeSubmitButton');
+  }
+
+  initializeDrawingTool(): void {
+    this.drawingTool = this.DrawService.initializeDrawingTool(
+      this.drawingToolId,
+      this.componentContent.stamps,
+      this.width,
+      this.height
+    );
+    this.initializeStudentData();
+    this.disableComponentIfNecessary();
+    this.setDrawingChangedListener();
+    this.setToolChangedListener();
+    this.setUpTools();
+
+    if (this.hasMaxSubmitCount() && this.hasSubmitsLeft()) {
+      this.isSubmitButtonDisabled = true;
+    }
+
+    if (this.isDisabled) {
+      this.drawingTool.canvas.removeListeners();
+    }
+  }
+
+  initializeStudentData(): void {
+    if (this.UtilService.hasShowWorkConnectedComponent(this.componentContent)) {
+      this.handleConnectedComponents();
+    } else if (
+      this.DrawService.componentStateHasStudentWork(this.componentState, this.componentContent)
+    ) {
+      this.setStudentWork(this.componentState);
+    } else if (this.UtilService.hasConnectedComponent(this.componentContent)) {
+      this.handleConnectedComponents();
+    } else if (this.isStarterDrawDataAvailable()) {
+      this.drawingTool.load(this.componentContent.starterDrawData);
+    }
+    if (
+      !this.DrawService.componentStateHasStudentWork(this.componentState, this.componentContent)
+    ) {
+      this.setAuthoredBackgroundIfNeeded();
+    }
+  }
+
+  setAuthoredBackgroundIfNeeded() {
+    if (this.isAuthoredBackgroundAvailable()) {
+      this.setAuthoredBackgroundAfterTimeout();
+    }
```

Comment:
We're checking  ```this.isAuthoredBackgroundAvailable()``` here and on line 133 in setAuthoredBackgroundIfAvailable(). Do we need to check in both places?